va/va_fool.c

```diff
@@ -290,7 +291,8 @@ static int va_FoolFillCodedBufEnc(struct fool_context *fool_ctx)
     }
     if (fd != -1) {
         fool_ctx->segbuf_enc = realloc(fool_ctx->segbuf_enc, file_stat.st_size);
-        read(fd, fool_ctx->segbuf_enc, file_stat.st_size);
+        size = read(fd, fool_ctx->segbuf_enc, file_stat.st_size);
+        assert(size == file_stat.st_size);
```

Comment:
instead of asserting it could be returning an error. 

ditto on the other places.

Reply:
Maybe give a warning message when size != file_stat.st_size, but need to add a new va_warnMessage() function in va.c.